src/journal/journald-syslog.c

```diff
@@ -145,6 +150,11 @@ void server_forward_syslog(Server *s, int priority, const char *identifier, cons
         if (LOG_PRI(priority) > s->max_level_syslog)
                 return;
 
+        if (s->facilities_store) {
+                if (!set_contains(s->facilities_store, UINT_TO_PTR(LOG_FAC(priority))))
+                        return;
+        }
+
```

Comment:
It seems that @poettering have explained that in https://github.com/systemd/systemd/pull/6361#issuecomment-315414364 and https://github.com/systemd/systemd/issues/5938.